tests/gitea_release.rs

```diff
@@ -0,0 +1,260 @@
+use std::{
+    fs::{copy, create_dir, read_to_string, write},
+    path::Path,
+};
+
+use helpers::*;
+use snapbox::{
+    assert_eq_path,
+    cmd::{cargo_bin, Command},
+};
+
+mod helpers;
+
+/// Run a `PreRelease` then `Release` for a repo not configured for gitea.
+///
+/// # Expected
+///
+/// Version should be bumped, and a new tag should be added to the repo.
+#[test]
+fn gitea_release() {
+    // Arrange.
+    let temp_dir = tempfile::tempdir().unwrap();
+    let temp_path = temp_dir.path();
+    let source_path = Path::new("tests/gitea_release");
+
+    init(temp_path);
+    commit(temp_path, "feat: Existing feature");
+    tag(temp_path, "v1.0.0");
+    commit(temp_path, "feat: New feature");
+
+    for file in ["knope.toml", "CHANGELOG.md", "Cargo.toml"] {
+        copy(source_path.join(file), temp_path.join(file)).unwrap();
+    }
+
+    // Act. Cannot run real release without integration testing gitea.
```

Comment:
Though I do wonder if I should set up some integration tests to run only right before release.

Reply:
mmmh... I think it would be enough to have a gitea instance (public or private) with a test repo.
The same would count for gh.

the way i did it is:
 I copied my own project (which was already using knope) into a test repo just to test knope releases, PRs and issues.